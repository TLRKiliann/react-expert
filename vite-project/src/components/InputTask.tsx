type PropsValue = {
	valueToRetrieve: {
		firstName: string,
		age: number,
		admin: boolean
	};
}

const InputTask = (props: PropsValue): JSX.Element => {
	return (
		<>
			<h1>Hello InputTask</h1>
			<h1 style={{color: "deepskyblue"}}>{props.valueToRetrieve.firstName}</h1>
			<h1 style={{color: "deepskyblue"}}>{props.valueToRetrieve.age}</h1>
			<h1 style={{color: "deepskyblue"}}>{props.valueToRetrieve.admin ? `Yes admin` : `not admin`}</h1>
		</>
	)
}

export default InputTask;