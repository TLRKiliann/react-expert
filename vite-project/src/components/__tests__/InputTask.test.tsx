import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { it, expect } from "vitest";
import InputTask from "../InputTask";
//import { handleValidate } from '../InputTask';

test('renders correctly InputTask', () => {
  const component = renderer.create(<InputTask />)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//props
//handleChangeInput
//handleValidate
