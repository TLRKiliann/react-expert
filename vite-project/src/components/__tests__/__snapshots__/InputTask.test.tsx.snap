// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly InputTask 1`] = `
[
  <h1>
    Hello InputTask
  </h1>,
  <h2
    style={
      {
        "color": "deepskyblue",
      }
    }
  />,
  <h2
    style={
      {
        "color": "deepskyblue",
      }
    }
  />,
  <h2
    style={
      {
        "color": "deepskyblue",
      }
    }
  >
    not admin
  </h2>,
  <label
    htmlFor="name"
  >
    Change firstname :
  </label>,
  <input
    data-testid="firstname"
    id="name"
    type="text"
  />,
  <p>
    Not changed
  </p>,
  <button
    data-testid="btntestvalidate"
    type="button"
  >
    Validate
  </button>,
]
`;
